scripts/cluster/agent.py

```diff
@@ -423,6 +443,285 @@ def configure():
+    output = subprocess.check_output("{}/microk8s-kubectl.wrapper get all --all-namespaces".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/status'.format(CLUSTER_API), methods=['POST', 'GET'])
+def status():
+    """
+    Web call to get the microk8s status
+    """
+    cmd = "{}/microk8s-status.wrapper --format yaml --timeout 60".format(snap_path)
+    if request.method.upper() == 'GET' and not is_extended_api_enabled():
+        error_msg = {"error": ext_api_error_message.format('GET')}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    if request.method.upper() == 'POST':
+        v = callback_token_validation(request)
+        if not v["valid"]: return v["response"]
+        if "addon" in request.json:
+            cmd = "{}/microk8s-status.wrapper -a {}".format(snap_path, request.json["addon"])
+
+    output = subprocess.check_output(cmd.split())
+
+    if request.json and "addon" in request.json:
+        json_output = {"addon": request.json["addon"], "status": output.decode().strip('\n')}
+    else:
+        json_output = yaml.full_load(output)
+
+    return app.response_class(response=json.dumps(json_output, sort_keys=False, indent=4), status=200, mimetype='application/json')
+
+
+@app.route('/{}/config'.format(CLUSTER_API), methods=['POST'])
+def config():
+    """
+    Web call to get the microk8s config
+    """
+    cmd = "{}/microk8s-config.wrapper".format(snap_path)
+    if not is_extended_api_enabled():
+        error_msg = {"error": ext_api_error_message.format('This')}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+
+    output = subprocess.check_output(cmd.split())
+    json_output = yaml.full_load(output)
+
+    return app.response_class(response=json.dumps(json_output, sort_keys=False, indent=4), status=200, mimetype='application/json')
+
+
+@app.route('/swagger.json', methods=['GET'])
+def swagger_json():
+    with open(os.path.join(os.path.dirname(__file__), "static/swagger.yaml")) as f:
+        content = f.read()
+    return app.response_class(response=json.dumps(yaml.full_load(content), sort_keys=False, indent=4), status=200, mimetype='application/json')
+
+
+def callback_token_validation(request):
+    if 'Content-Type' in request.headers and request.headers['Content-Type'] == 'application/json':
+        callback_token = request.json['callback']
+    elif 'Callback-Token' in request.headers:
+        callback_token = request.headers['Callback-Token']
+    else:
+        callback_token = request.form['callback']
+    valid = is_valid(callback_token, callback_token_file)
+    resp = None
+    if not valid:
+        error_msg = {"error": "Invalid token"}
+        resp = Response(json.dumps(error_msg), mimetype='application/json', status=500)
+    return {"valid": valid, "response": resp}
+
+
+def is_extended_api_enabled():
```

Comment:
Same, here does this need to be part of the MicroDash PR? I understand this is a flag to enable/disable certain functionalities. I think this belongs into its own PR so we discuss on what we want to feature-flag and how. Thank you.

Reply:
will do in a separate PR